data/account/participation.go

```diff
@@ -217,12 +217,9 @@ func FillDBWithParticipationKeys(store db.Accessor, address basics.Address, firs
 	}
 	// Persist the Participation into the database
 	err = part.Persist()
-	if err != nil {
-		return PersistedParticipation{}, err
+	if err == nil {
+		err = stateProofSecrets.Persist() // must be called after part.Persist()
 	}
-
-	err = stateProofSecrets.Persist() // must be called after part.Persist() !
-
 	return part, err
```

Comment:
notice here, without my change, in the case of stateProofSecrets.Persist returning an error, it is still returning part with err not nil. 
If we want to return PersistedParticipation{} in case of an error, we net to check err in two places. I think this will unnecessarily complicate the code.